File: Days-lived/days_lived.py
from datetime import date
while True:
    name = str(input("Enter Your Name: "))

    if name == "0":
        print("You Have Exited The Program")
        break

    def get_birth_date():
        day = int(input("Enter the day of your birth: "))
        month = int(input("Enter the month of your birth: "))
        year = int(input("Enter the year of your birth: "))

        return date (year, month, day)

    def calculate_days(birth_date):
        today = date.today()
        days_passed = (today - birth_date). days
        return days_passed

    def format_date(date_obj):
        return date_obj.strftime("%d %B %Y")

    # Print the result
    birth_date = get_birth_date()
    formatted_birth_date = format_date(birth_date)
    days_passed = calculate_days(birth_date)
    print("-" * 50)
    print(
        f"kamu lahir pada '{formatted_birth_date}' & telah hidup salam {days_passed} hari, {name}!"
    )

File: Days-lived/days_lived.cpp
#include <iostream>
#include <ctime>
#include <iomanip>
using namespace std;

struct date
{
    int day, month, year;
};

string getFormattedDate(date dateObj)
{
    struct tm timeStruct = {0};
    timeStruct.tm_year = dateObj.year - 1900;
    timeStruct.tm_mon = dateObj.month - 1;
    timeStruct.tm_mday = dateObj.day;

    char buffer[80];
    strftime(buffer, sizeof(buffer), "%d %B %Y", &timeStruct);
    return buffer;
}

date getBirthDate()
{
    date birthDate;
    cout << "Enter the day of your birth: ";
    cin >> birthDate.day;
    cout << "Enter the month of your birth: ";
    cin >> birthDate.month;
    cout << "Enter the year of your birth: ";
    cin >> birthDate.year;
    return birthDate;
}

int calculateDays(date birthDate)
{
    time_t now = time(0);
    tm *today = localtime(&now);
    tm birthTime = {0};
    birthTime.tm_year = birthDate.year - 1900;
    birthTime.tm_mon = birthDate.month - 1;
    birthTime.tm_mday = birthDate.day;

    double seconds = difftime(now, mktime(&birthTime));
    int daysPassed = static_cast<int>(seconds / (60 * 60 * 24));

    return daysPassed;
}

int main()
{
    string name;
    cout << "Enter Your name: ";
    getline(cin, name);

    date birthDate = getBirthDate();
    string formattedBirthDate = getFormattedDate(birthDate);
    int daysPassed = calculateDays(birthDate);

    cout << string(50, '-') << endl;
    cout << "Kamu lahir pada " << formattedBirthDate << " & telah hidup selama " << daysPassed << " hari, " << name << "!" << endl;

    return 0;
}